src/ray/raylet/lineage_cache.cc

```diff
@@ -286,12 +294,22 @@ void PopAncestorTasks(const UniqueID &task_id, Lineage &lineage) {
 }
 
 void LineageCache::HandleEntryCommitted(const UniqueID &task_id) {
+  RAY_LOG(DEBUG) << "task committed: " << task_id;
   auto entry = lineage_.PopEntry(task_id);
   for (const auto &parent_id : entry->GetParentTaskIds()) {
     PopAncestorTasks(parent_id, lineage_);
   }
+  // Mark this task as COMMITTED. Any tasks that were dependent on it and are
+  // ready to be written may now be flushed to the GCS.
   RAY_CHECK(entry->SetStatus(GcsStatus_COMMITTED));
   RAY_CHECK(lineage_.SetEntry(std::move(*entry)));
+
+  // Stop listening for notifications about this task.
+  auto it = subscribed_tasks_.find(task_id);
+  if (it != subscribed_tasks_.end()) {
+    task_pubsub_.CancelNotifications(JobID::nil(), task_id, client_id_);
```

Comment:
It seems like for this use case, it would be fine for the `Cancel` to happen automatically once the GCS sends the notification, right?

The issue with that is that it involves special-casing the behavior in the redis module?

Reply:
Hmm yeah, for the `Table` interface, we could just call `Cancel` as soon as an entry is found since table entries are supposed to be immutable. Should I change that in this PR?

Reply:
It can be a subsequent PR if that makes sense.